components/dashboard/src/data/prebuilds/latest-project-prebuild-query.ts

```diff
@@ -16,13 +16,15 @@ type Args = {
 export const useLatestProjectPrebuildQuery = ({ projectId }: Args) => {
     return useQuery<LatestProjectPrebuildQueryResult>({
         queryKey: getLatestProjectPrebuildQueryKey(projectId),
+        // Prevent bursting for latest project prebuilds too frequently
+        staleTime: 1000 * 60 * 1, // 1 minute
         queryFn: async () => {
             const latestPrebuilds = await getGitpodService().server.findPrebuilds({
                 projectId,
                 latest: true,
             });
 
-            return latestPrebuilds[0];
+            return latestPrebuilds[0] || null;
```

Comment:
Curious: What's the reason to use `null` over `undefined`? :thinking:

Reply:
It's a constraint [react query imposes](https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#undefined-is-an-illegal-cache-value-for-successful-queries) for query results - so in this case `null` becomes an explicit value that represents no result, and conforms to the libraries constraints.